src/algorithms/all_paths.c

```diff
@@ -62,8 +63,6 @@ static void _addNeighbors(AllPathsCtx *ctx, LevelConnection *frontier, uint32_t
 
 AllPathsCtx *AllPathsCtx_New(Node *src, Node *dst, Graph *g, int *relationIDs, int relationCount,
 							 GRAPH_EDGE_DIR dir, unsigned int minLen, unsigned int maxLen) {
-	assert(src);
```

Comment:
Why did you removed this line?

Reply:
allow this op to be part of optional match flow

Reply:
Please add test for OPTIONAL VAR LEN traversal

Reply:
`ctx->path = Path_New(1);` might need to change if `src` is NULL.